### PR TITLE
Add DomainNotFound error handling and missing documentation

### PR DESCRIPTION
**Pull request by akrishna90**
## Is there a related GitHub Issue?
#161 

## What is this change about?
Add missing edge case handling and documentation

- throws a HTTP 404 instead of a HTTP 500 when Domain is not found
- add documentation for setting default domains

## Does this PR introduce a breaking change?
No

## Acceptance Steps
See #161

### CODE REVIEW

File: api/apis/space_manifest_handler_test.go [hunk 1]
```diff
@@ -310,6 +310,16 @@ var _ = Describe("SpaceManifestHandler", func() {
 				Expect(payload.Applications[0].DefaultRoute).To(BeTrue())
 			})
 		})
+
+		When("applying the manifest errors with NotFoundErr", func() {
+			BeforeEach(func() {
+				applyManifestAction.Returns(repositories.NotFoundError{})
+			})
+
+			It("respond with Unknown Error", func() {
```
Comment (tcdowney) [inline api/apis/space_manifest_handler_test.go:319]: I think this should be "Not Found Error" instead.

**Comment by tcdowney** [comment]
I know the original story called for a `404`, but I think in this case it might make sense to do handle this differently than other "not found" cases and return a `422` saying that there is no default domain configured or something.

Because most cases when a user is getting a `404` they're trying to view or do something with a specific resource, but in this case all they're doing is including the `default-domain: true` key in their app manifest and I think a `404` might be confusing. The `POST /v3/spaces/:guid/actions/apply_manifest` endpoint given what it's doing.

**Comment by tcdowney** [comment]
In some ways this flow is easier to reason about in CF for VMs cause it does most of the work as part of an async job and its just able to set the job status to `FAILED` and build up a list of errors [on the job](https://v3-apidocs.cloudfoundry.org/version/3.115.0/index.html#get-a-job) to give the user.

**Comment by akrishna90** [comment]
Thanks @tcdowney for the feedback. I will make the change to return a `422` 
cc @Birdrock (author of the story)

**Review by tcdowney — APPROVED** [review]
LGTM! Seems fine to special case it right now since we've got https://github.com/cloudfoundry/cf-k8s-controllers/issues/699 to improve errors here more broadly.

**Pull request merged**